packages/jats-xml/src/cli/parse.ts

```diff
@@ -0,0 +1,201 @@
+import { Command } from 'commander';
+import fs from 'fs';
+import { extname } from 'path';
+import { clirun, isUrl, tic, writeFileToFolder } from 'myst-cli-utils';
+import checkdoi from 'doi-utils';
+import chalk from 'chalk';
+import { getSession } from '../session';
+import type { ISession } from '../types';
+import { Tags } from '../types';
+import { Jats } from '../jats';
+import { toText } from 'myst-common';
+import { select, selectAll } from 'unist-util-select';
+import { downloadJatsFromUrl } from '../download';
+import { DEFAULT_RESOLVERS } from '../resolvers';
+import { findDoi, formatDate, toDate } from '../utils';
+
+function hasValidExtension(output: string) {
+  return ['.xml', '.jats'].includes(extname(output).toLowerCase());
+}
+
+async function downloadJats(session: ISession, urlOrDoi: string, output: string): Promise<string> {
+  if (fs.existsSync(urlOrDoi)) {
+    throw new Error(`File "${urlOrDoi}" is local and cannot be downloaded!`);
+  }
+  if (!(checkdoi.validate(urlOrDoi) || isUrl(urlOrDoi))) {
+    throw new Error(`Path must be a URL or DOI, not "${urlOrDoi}"`);
+  }
+  if (!hasValidExtension(output)) {
+    session.log.warn(
+      `The extension ${extname(
+        output,
+      )} is not a valid extension for JATS, try using ".xml" or ".jats"`,
+    );
+  }
+  const data = await downloadJatsFromUrl(session, urlOrDoi, DEFAULT_RESOLVERS);
+  writeFileToFolder(output, data);
+  return data;
+}
+
+async function parseJats(session: ISession, file: string): Promise<Jats> {
+  const toc = tic();
+  if (fs.existsSync(file)) {
+    session.log.debug(`Found ${file} locally, parsing`);
+    const data = fs.readFileSync(file).toString();
+    session.log.debug(toc(`Parsed JATS file from disk in %s`));
+    return new Jats(data);
+  }
+  if (checkdoi.validate(file) || isUrl(file)) {
+    const data = await downloadJatsFromUrl(session, file, DEFAULT_RESOLVERS);
+    session.log.debug(toc(`Downloaded and parsed JATS file in %s`));
+    return new Jats(data);
+  }
+  throw new Error(`Could not find ${file} locally, and it doesn't look like a URL or DOI`);
+}
+
+function formatLongString(data: string, offset = 0, length = 88 - offset): string {
```

Comment:
These utilities are maybe throw away or something we can bring to `myst-cli-utils`. Might be worth looking into some slightly better CLI tools for formatting.